### PR TITLE
Adds a new field to Course model

### LINKED ISSUE #49: add field to distinguish resources from courses
Items in the /RES/ folder are OCW resources and not true courses. We would like to flag this on import so that they can potentially be indexed differently. That being said, they should have all the same fields, including instructors and semester.

### PR DESCRIPTION
**Pull request by alizag**
#### What are the relevant tickets?
closes #49 

#### What's this PR do?
Adds `is_resource` field to Course model.  Also sets `is_resource` flag to `True` if the course being processed is a resource.

### CODE REVIEW

**Comment by Ferdi** [comment]
We should make this a bit more generic, for example
learning_resource_type with possible values of course, ocw_learbing_resource and maybe later, video etc

File: course_catalog/tasks_helpers.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def safe_load_json(json_string, json_file_key):
         return {}
 
 
-def digest_ocw_course(master_json, last_modified, course_instance, is_published):
+def digest_ocw_course(master_json, last_modified, course_instance, is_published, course_prefix=""):
```
Comment (mbertrand) [inline course_catalog/tasks_helpers.py]: Add ` . #pylint:disable=too-many-locals` to pass tests

Comment (alizag) [reply course_catalog/tasks_helpers.py]: Addressed in [49c6574](https://github.com/mitodl/course-catalog/pull/50/commits/49c65746f965ee9886722f59157e3aee01072eb8)

**Comment by alizag** [comment]
@Ferdi comment addressed in [b324372](https://github.com/mitodl/course-catalog/pull/50/commits/b32437293f1f4f78a2b1033d6d1bd074f2b3868b)

File: course_catalog/views.py [hunk 2]
```diff
@@ -38,14 +38,16 @@ def ocw_course_report(request):
     """
     Returns a JSON object reporting OCW course sync statistics
     """
-    ocw_courses = Course.objects.filter(platform=PlatformType.ocw.value)
+    ocw_courses = Course.objects.filter(platform=PlatformType.ocw.value, is_resource=False)
```
Comment (mbertrand) [inline course_catalog/views.py]:
```python
Course.objects.filter(platform=PlatformType.ocw.value, learning_resource_type=ResourceType.course.value)
```

File: course_catalog/views.py [hunk 3]
```diff
@@ -38,14 +38,16 @@ def ocw_course_report(request):
...
     published_ocw_courses_with_image = ocw_courses.filter(published=True, image_src__isnull=False).count()
     unpublished_ocw_courses = ocw_courses.filter(published=False).count()
     ocw_courses_without_image = ocw_courses.filter(image_src="").count()
+    ocw_resources = Course.objects.filter(platform=PlatformType.ocw.value, is_resource=True).count()
```
Comment (mbertrand) [inline course_catalog/views.py]:
```python
ocw_resources = Course.objects.filter(
    platform=PlatformType.ocw.value, 
    learning_resource_type=ResourceType.ocw_resource.value
)
```

**Pull request merged**